### PR TITLE
SMHE-729: Add TEST_ALARM_SCHEDULER message

### PR DESCRIPTION
**Pull request by abolder**
Signed-off-by: Andre Bolder <andre.bolder@alliander.com>

Added schedule test alarms operation for the alarms: PHASE_OUTAGE_TEST (fom smr5) and LAST_GASP_TEST (for smr55).
Added to:

- osgp-adapter-ws-smartmetering
- osgp-core
- osgp-protocol-adapter

### CODE REVIEW

File: osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/application/mapping/AdhocMapper.java [hunk 1]
```diff
@@ -21,5 +23,11 @@ public void configure(final MapperFactory mapperFactory) {
     mapperFactory.getConverterFactory().registerConverter(new XsdDateTimeToLongConverter());
     mapperFactory.getConverterFactory().registerConverter(new CosemObisCodeConverter());
     mapperFactory.getConverterFactory().registerConverter(new ObisCodeValuesConverter());
+
+    mapperFactory
+        .classMap(ResponseData.class, TestAlarmSchedulerResponse.class)
+        .field("resultType", "result")
+        .field("messageData", "description")
+        .register();
```
Comment (kroesctrl) [inline osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/application/mapping/AdhocMapper.java:31]: +1 this should have been done in other request/responses too

File: osgp/platform/osgp-adapter-ws-smartmetering/src/test/java/org/opensmartgridplatform/adapter/ws/smartmetering/application/mapping/AdhocMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    final org.opensmartgridplatform.domain.core.valueobjects.smartmetering
+            .TestAlarmSchedulerRequestData
+        mapped =
+            this.adhocMapper.map(
+                testAlarmSchedulerRequestData,
+                org.opensmartgridplatform.domain.core.valueobjects.smartmetering
+                    .TestAlarmSchedulerRequestData.class);
+
+    Assertions.assertThat(mapped)
+        .hasToString(
```
Comment (kroesctrl) [inline osgp/platform/osgp-adapter-ws-smartmetering/src/test/java/org/opensmartgridplatform/adapter/ws/smartmetering/application/mapping/AdhocMapperTest.java]: I would like to see also test of specific fields and not only te toString

File: osgp/platform/osgp-core/src/main/resources/db/migration/V202209051216124523__Adds_SMR_5_5_protocol_info.sql [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    protocol,
+    protocol_version,
+    outgoing_requests_property_prefix,
+    incoming_responses_property_prefix,
+    incoming_requests_property_prefix,
+    outgoing_responses_property_prefix,
+    parallel_requests_allowed)
+VALUES (
+    '2021-10-01 00:00:00',
+    '2021-10-01 00:00:00',
```
Comment (kroesctrl) [inline osgp/platform/osgp-core/src/main/resources/db/migration/V202209051216124523__Adds_SMR_5_5_protocol_info.sql]: 2022

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/TestAlarmSchedulerRequestData.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import org.opensmartgridplatform.domain.core.valueobjects.DeviceFunction;
+import org.opensmartgridplatform.shared.exceptionhandling.ComponentType;
+import org.opensmartgridplatform.shared.exceptionhandling.FunctionalException;
+import org.opensmartgridplatform.shared.exceptionhandling.FunctionalExceptionType;
+import org.opensmartgridplatform.shared.exceptionhandling.OsgpException;
+
+@Getter
+@AllArgsConstructor
```
Comment (kroesctrl) [inline osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/TestAlarmSchedulerRequestData.java]: AllArgsConstructor is a annotation i do not like personally, change of order of attributes of same type changes the constructor. Consider to make constructor explicit

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/dlmsobjectconfig/DlmsObjectConfigSmr55.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2021 Alliander N.V.
```
Comment (kroesctrl) [inline osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/dlmsobjectconfig/DlmsObjectConfigSmr55.java]: 2022

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/entities/Protocol.java [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2019 Smart Society Services B.V.
+ * Copyright 2022 Smart Society Services B.V.
```
Comment (kroesctrl) [inline osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/entities/Protocol.java]: revert to 2019

**Review by kroesctrl — CHANGES_REQUESTED** [review]

**Comment by kroesctrl** [comment]
You should trigger the nightly build in this PR and post a link in a comment to the succeeded build in this PR, because you added @NightlyBuildOnly tests

File: osgp/platform/osgp-adapter-ws-smartmetering/src/test/java/org/opensmartgridplatform/adapter/ws/smartmetering/application/mapping/AdhocMapperTest.java (hunk 2 above)
Comment (abolder) [reply osgp/platform/osgp-adapter-ws-smartmetering/src/test/java/org/opensmartgridplatform/adapter/ws/smartmetering/application/mapping/AdhocMapperTest.java]: Yes, that would be nicer. I will change this.

File: osgp/platform/osgp-core/src/main/resources/db/migration/V202209051216124523__Adds_SMR_5_5_protocol_info.sql (hunk 3 above)
Comment (abolder) [reply osgp/platform/osgp-core/src/main/resources/db/migration/V202209051216124523__Adds_SMR_5_5_protocol_info.sql]: Changed to 2022

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/TestAlarmSchedulerRequestData.java (hunk 4 above)
Comment (abolder) [reply osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/TestAlarmSchedulerRequestData.java]: Added an explicit constructor and removed the AllArgsConstructor annotation.

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/dlmsobjectconfig/DlmsObjectConfigSmr55.java (hunk 5 above)
Comment (abolder) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/dlmsobjectconfig/DlmsObjectConfigSmr55.java]: Changed to 2022

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/entities/Protocol.java (hunk 6 above)
Comment (abolder) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/entities/Protocol.java]: reverted to 2019

**Comment by abolder** [comment]
New Jenkins server GXF Nightly : https://jenkins.fdp.osgp.cloud/view/Nightly%20Build/job/gxf-nightly-build/462/

**Comment by abolder** [comment]
✘ New GXF Nightly build after https://github.com/SmartSocietyServices/release/pull/289 merge. 
https://jenkins.fdp.osgp.cloud/view/Nightly%20Build/job/gxf-nightly-build/463/

**Comment by abolder** [comment]
✘ https://jenkins.fdp.osgp.cloud/view/Nightly%20Build/job/gxf-nightly-build/467/

**Comment by abolder** [comment]
✘ https://jenkins.fdp.osgp.cloud/view/Nightly%20Build/job/gxf-nightly-build/468/

**Comment by abolder** [comment]
✘ https://jenkins.fdp.osgp.cloud/view/Nightly%20Build/job/gxf-nightly-build/469/

**Comment by abolder** [comment]
https://jenkins.fdp.osgp.cloud/view/Nightly%20Build/job/gxf-nightly-build/469/

**Comment by abolder** [comment]
✘ https://jenkins.fdp.osgp.cloud/view/gxf/job/gxf-nightly-build/474/

**Comment by abolder** [comment]
✔︎ Merged development branch into this one to fix the ExchangeUserKeyOnMbusDevice cucumber test.

https://jenkins.fdp.osgp.cloud/view/gxf/job/gxf-nightly-build/476/

**Review by kroesctrl — APPROVED** [review]
Approved, when nightlies are succesful

**Comment by abolder** [comment]
Nightlies are succesfull and added to the checklist

**Pull request merged**